src/tests/asyncTests.ts

```diff
@@ -364,12 +367,56 @@ export class AsyncTests {
       const queryPromises = queries.map(async (query) => {
         return queryAll(connection, query, true);
       });
-      const apexTestResults = await Promise.all(queryPromises);
-      return apexTestResults as ApexTestResult[];
+      const testResults = await Promise.all(queryPromises);
+      if (isFlowRunTest) {
+        return this.convertFlowTestResult(
+          testResults as FlowTestResult[]
+        ) as ApexTestResult[];
+      }
+      return testResults as ApexTestResult[];
     } finally {
       HeapMonitor.getInstance().checkHeapSize('asyncTests.getAsyncTestResults');
     }
   }
+  /**
+   * @returns Convert FlowTest result to ApexTestResult type
+   */
+
+  public convertFlowTestResult(
```

Comment:
This is overloading the ApexTestResult and forcing it to comply with the type. This breaks the contract we have with other consumers.